webapp/channels/src/utils/desktop_api.ts

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import semver from 'semver';
+
+import {isDesktopApp} from 'utils/user_agent';
+
+// TODO: Temporary typings
+import type {DesktopAPI} from '../../../../../desktop/src/types/api/main';
+declare global {
+    interface Window {
+        desktopAPI?: DesktopAPI;
+    }
+}
+
+class DesktopApp {
```

Comment:
Heh, I kinda wanted your help on this actually. Not sure the best way to break up this module, since most of the API calls don't rely on that class (except in the legacy case :P) so I just left it the way I originally named it.

Open to suggestions :)